MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewControllerExtension.swift

```diff
@@ -70,32 +70,141 @@ extension TransactionsTableViewController {
+        var message = ""
+        var contact: Contact?
+        var microTari: MicroTari?
+        var direction: TransactionDirection = .inbound
+
+        //If it's the first group and we're showing the pending group
+        if indexPath.section == 0 && showPendingGroup {
+            if indexPath.row < pendingInboundTransactions.count {
+                let inboundTransaction = pendingInboundTransactions[indexPath.row]
+
+                message = inboundTransaction.message.0
+                contact = inboundTransaction.contact.0
+                microTari = inboundTransaction.microTari.0
+                direction = inboundTransaction.direction
+            } else {
+                let outboundTransaction = pendingOutboundTransactions[indexPath.row - pendingInboundTransactions.count]
+
+                message = outboundTransaction.message.0
+                contact = outboundTransaction.contact.0
+                microTari = outboundTransaction.microTari.0
+                direction = outboundTransaction.direction
+            }
+        } else {
+            //Handle as a completed transaction
+            let index = showPendingGroup ? indexPath.section - 1 : indexPath.section
+
+            let transaction = groupedCompletedTransactions[index][indexPath.row]
+
+            message = transaction.message.0
+            contact = transaction.contact.0
+            microTari = transaction.microTari.0
+            direction = transaction.direction
+        }
+
+        cell.icon.image = Theme.shared.images.transfer
 
-        cell.icon.image = transaction.icon
-        cell.userNameLabel.text = transaction.userName
-        cell.descriptionLabel.text = transaction.description
+        var userName = "-"
+        if let c = contact {
+            let (alias, aliasError) = c.alias
+            userName = alias
+        }
 
-        cell.setValueLabel(tariValue: transaction.value)
+        cell.userNameLabel.text = userName
+        cell.setMessage(message != "" ? message : "*Missing note*")
+        cell.setValueLabel(microTari: microTari != nil ? microTari! : MicroTari(0), direction: direction)
 
         return cell
     }
+
+    func setEmptyView() {
+        let emptyView = UIView(frame: CGRect(x: tableView.center.x, y: tableView.center.y, width: tableView.bounds.size.width, height: tableView.bounds.size.height))
+
+        let imageView = UIImageView(image: Theme.shared.images.emptyStateTransactionList)
+        imageView.frame = CGRect(x: 0, y: 0, width: 275, height: 192)
+        imageView.contentMode = .scaleAspectFit
+        imageView.center = CGPoint(x: tableView.center.x, y: tableView.center.y - 200)
+        emptyView.addSubview(imageView)
+
+        let titleLabel = UILabel()
+        let messageLabel = UILabel()
+        messageLabel.numberOfLines = 0
+        messageLabel.textAlignment = .center
+        titleLabel.text = NSLocalizedString("Let's get started", comment: "Home view table when there are no transactions")
```

Comment:
I normally use key values like "home.lets_get_started", I think they're less prone to typos (upper/lower, non-alphanumeric chars etc.), but this is just my idea.

Reply:
Yeah good idea, I'll make an issue for this to come back to. With limited time it'll be easier to not have to keep a language file up to date yet.